clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -278,12 +278,23 @@ protected void onJoinComplete(int generation,
      *
      * @param now current time in milliseconds
      */
-    public void poll(long now) {
+    public void poll(long now, long remainingMs) {
         invokeCompletedOffsetCommitCallbacks();
 
-        if (subscriptions.partitionsAutoAssigned() && coordinatorUnknown()) {
-            ensureCoordinatorReady();
-            now = time.milliseconds();
+        if (subscriptions.partitionsAutoAssigned()) {
+            if (coordinatorUnknown()) {
+                ensureCoordinatorReady();
+                now = time.milliseconds();
+            }
+        } else {
+            // For manually assigned partitions, if there are no ready nodes, await metadata
+            // to avoid tight polling loop with no channels
+            if (metadata.updateRequested() && !client.hasReadyNodes()) {
```

Comment:
The objects being checked are in `ConsumerCoordinator`. Rather than add a method to the coordinator to do this, it made sense to do the check here, close to where `ensureCoordinatorReady` was being called, which invokes `awaitMetadataUpdate` for this scenario when group management is used.